apps/personal/contacts/main.py

```diff
@@ -4,57 +4,113 @@
         Listbox(i=self.i, o=self.o, contents=contact_attrs).activate()
 
-
-def find_contact_files(folder):
-    # type: (str) -> list(str)
-    home = os.path.expanduser(folder)
-    if not os.path.exists(home):
-        os.mkdir(home)
-    contact_card_files = [os.path.join(home, f) for f in os.listdir(home) if f.lower().endswith("vcf")]
-    return contact_card_files
-
-
-def load_vcf(folder):
-    # type: (str) -> None
-    contact_card_files = find_contact_files(folder)
-    contacts = VCardContactConverter.from_vcards(contact_card_files)
-
-    address_book = AddressBook()
-    for contact in contacts:
-        address_book.add_contact(contact)
-    address_book.save_to_file()
-    logger.info("Saved to {}".format(address_book.get_save_file_path()))
-
+    def open_actions_menu(self):
+        menu_contents = [
+            ["Settings", lambda: self.open_settings_page()],
+            ["CardDAV Import", lambda: self.synchronize_carddav()],
+            ["Reset address book", lambda: self.reset_addressbook()]
+        ]
+        Menu(menu_contents, i=self.i, o=self.o, name="My menu").activate()
+
+    def open_settings_page(self):
+        if self.vdirsyncer_executable:
+            vdirsyncer_executable = self.vdirsyncer_executable
+        else:
+            vdirsyncer_executable = 'Not found'
+
+        attrs = [
+            ["-- VdirSyncer"],
+            ["Executable: {}".format(vdirsyncer_executable) ],
+            ["Config: {}".format(VDIRSYNCER_CONFIG)]
+        ]
+        Listbox(i=self.i, o=self.o, contents=attrs).activate()
+
+    def reset_addressbook(self):
+        alert = "This action will delete all of your contacts. Are you sure?"
+        do_reset = DialogBox('yc', i=self.i, o=self.o, message=alert,
+                             name="Address book reset").activate()
+
+        if do_reset:
+            self.address_book.reset()
+            announce = "All of your contacts were deleted."
+            Printer(announce, i=self.i, o=self.o, sleep_time=2, skippable=True)
+            # Reload the now empty address book
+            self.reload()
+
+    def synchronize_carddav(self):
+        if (not os.path.isfile(self.vdirsyncer_executable) or
+        not os.access(self.vdirsyncer_executable, os.X_OK)):
+            Printer('Could not execute vdirsyncer.', i=self.i, o=self.o,
+                    sleep_time=2, skippable=True)
+            callback()
+            return;
+
+        vdirsyncer_command = "{} -c {} sync contacts".format(
+            self.vdirsyncer_executable, VDIRSYNCER_CONFIG
+        )
+        logger.info("Calling vdirsyncer to synchronize contacts")
+        with LoadingIndicator(self.i, self.o, message="Syncing contacts"):
+            exit_status = os.system(vdirsyncer_command)
+
+        if (exit_status != 0):
+            error_msg = "Error in contact synchronization. Did you configure \
+            vdirsyncer?"
+            Printer(error_msg, i=self.i, o=self.o, sleep_time=3,
```

Comment:
`i` and `o` aren't keyword arguments, so, this looks like code that will throw an exception. Ditto for all other `Printer` invocations. Also, `skippable` is `True` by default - so, the shortest and best way to use `Printer` in your case is like this:
`Printer("text", self.i, self.o, 3)`,  Furthermore, I suggest you use `PrettyPrinter` for long text - this won't print the text fully. The simplest change to fix this is probably adding `from ui import PrettyPrinter as Printer` at the top =)

I apologize if this was taken from some other place, not all places in ZPUI are up-to-date - see #113 =(

Reply:
Fixed by bb19598.